### PR TITLE
feat: integration of starting overview page

### PR DESCRIPTION
**Pull request by Teja2045**
#879

### CODE REVIEW

File: frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  useEffect(() => {
+    chainIDs.forEach((chainID) => {
+      const chainInfo = networks[chainID]?.network;
+      const address = networks[chainID]?.walletInfo?.bech32Address;
+      //   const denom =
+      //     networks?.[chainID]?.network?.config?.currencies?.[0]?.coinMinimalDenom;
+
+      dispatch(
+        getBalances({
+          baseURL: chainInfo.config.rest,
```
Comment (charymalloju) [inline frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx:217]: When calling a function, the same object is being created twice.

File: frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+      chainIdToNames[nameToChainIDs[key]] = key;
+    }
+    return chainIdToNames;
+  }, [nameToChainIDs]);
+
+  const convertToDollars = (denom: string, amount: number) => {
+    // todo: after common slice
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    const tokensPriceInfo: any = {};
+    if (tokensPriceInfo?.[denom]) {
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx:44]: tokens prices are not integrated yet... (common slice).

File: frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
+'use client';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { RootState } from '../../../..//store/store';
```
Comment (aleem1314) [inline frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx:3]: ### 
```suggestion
import { RootState } from '../../../../store/store';
```

File: frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    if (tokensPriceInfo?.[denom]) {
+      const price = +tokensPriceInfo?.[denom]?.info?.[USD_CURRENCY] || 0;
+      return amount * price;
+    }
+    return 0;
+  };
+
+  const getDenomInfo = useCallback(
+    (chainID: string): { denom: string; decimals: number } => {
+      const currency = networks?.[chainID]?.network?.config?.currencies?.[0];
```
Comment (aleem1314) [inline frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx:54]:
```suggestion
      const currency = networks.[chainID].network.config.currencies.[0];
```

Ditto here. If chain information exists, then currency also exists, as we enforce this in the network details object.

File: frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+          );
+      }
+      sortedChains = [...sortedChains, chain];
+    });
+    sortedChains.sort((x, y) => y.usdValue - x.usdValue);
+    return sortedChains.map((chain) => chain.chainID);
+  }, [chainIDs, getDenomInfo, balanceChains]);
+
+  const getIBCSortedChainIds = useCallback(() => {
+    let sortedIBCChains: {
```
Comment (aleem1314) [inline frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx:173]: Can we move all these parsing and transforming function into another file? Let's keep JSX pages small.

File: frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    }
+    return chainIdToNames;
+  }, [nameToChainIDs]);
+
+  const convertToDollars = (denom: string, amount: number) => {
+    // todo: after common slice
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+    const tokensPriceInfo: any = {};
+    if (tokensPriceInfo?.[denom]) {
+      const price = +tokensPriceInfo?.[denom]?.info?.[USD_CURRENCY] || 0;
```
[the remark this answers is not in the document]
Comment (Teja2045) [reply frontend/src/app/(routes)/(overview)/overview-components/OverviewPage.tsx:38]: Initially the token prices are empty (before fetching from api). If we try fetch price before api retured some value, wouldn't that throw undefined error?

same for a lot of other slices like staking (feegrant, distribution, etc...) where useselector might throw error that if that specific chain data is not set...

**Review by aleem1314 — APPROVED** [review]
Pre-approving. Please link relevant issue in the PR description.

**Pull request merged**